compiler/src/org.graalvm.compiler.replacements/src/org/graalvm/compiler/replacements/IntrinsicGraphBuilder.java

```diff
@@ -134,12 +146,46 @@ protected IntrinsicGraphBuilder(OptionValues options, DebugContext debug, CorePr
                 FixedWithNextNode fixedWithNextNode = (FixedWithNextNode) fixedNode;
                 assert fixedWithNextNode.next() == null : "cannot append instruction to instruction which isn't end";
                 lastInstr = fixedWithNextNode;
+
+            } else if (fixedNode instanceof WithExceptionNode) {
+                WithExceptionNode withExceptionNode = (WithExceptionNode) fixedNode;
+                AbstractBeginNode normalSuccessor;
+                if (withExceptionNode instanceof SingleMemoryKill) {
+                    normalSuccessor = graph.add(KillingBeginNode.create(((SingleMemoryKill) withExceptionNode).getKilledLocationIdentity()));
+                } else if (withExceptionNode instanceof MultiMemoryKill) {
+                    normalSuccessor = graph.add(MultiKillingBeginNode.create(((MultiMemoryKill) withExceptionNode).getKilledLocationIdentities()));
+                } else {
+                    assert !(withExceptionNode instanceof MemoryKill) : "missing a MemoryKill";
+                    normalSuccessor = graph.add(new BeginNode());
+                }
```

Comment:
This should just be:
```suggestion
                AbstractBeginNode normalSuccessor = withExceptionNode.createNextBegin();
```
Looks good to me otherwise.

Reply:
Thanks, didn't know that method exists. But I'm glad it does.